src/rust/ide/ast/impl/src/lib.rs

```diff
@@ -1058,16 +1058,39 @@ impl Ast {
         Ast::new(var,Some(id))
     }
 
+    /// Creates an AST node with `Var` shape.
     pub fn var<Str:ToString>(name:Str) -> Ast {
         let var = Var{name:name.to_string()};
         Ast::from(var)
     }
 
+    /// Creates an AST node with `Opr` shape.
     pub fn opr<Str:ToString>(name:Str) -> Ast {
         let opr = Opr{name:name.to_string() };
         Ast::from(opr)
     }
 
+    /// Creates an AST node with `SectionLeft` shape.
+    pub fn section_left<Arg:Into<Ast>, Opr:Into<Ast>>(arg:Arg, opr:Opr) -> Ast {
```

Comment:
Here — and in other cases — I'd accept `opr:impl Str` and then pack it into Opr node with Ast::opr.
Having shape other thatn `Opr` in these Infix-like shapes is illegal, shouldn't be possible to build.  (unfortunately can't quire prevent this with data structures as they are defined)

Reply:
@mwu-tow I have added the constraint `Opr:ToString` just like in the other definitions.

Reply:
The other definitions are outdated code. The preferred approach is using `Str` which is a unified way to express string-like generic type.